### PR TITLE
[docs] state compression and compressed nfts

### PR DESCRIPTION
**Pull request by nickfrosty**
#### Problem

Docs do not include info on State Compression and/or compressed NFTs.

#### Summary of Changes

- added doc for State compression
- added guide for compressed NFTs
- updated the sidebar to link to these new docs

### CODE REVIEW

**Comment by ngundotra** [comment]
Can this be reconciled with https://spl.solana.com/account-compression please? I think the table you've written up is definitely new & useful, but can you also link to your calculator website? I think that's super useful.

Also, why the triple `:::` in certain places? Doesn't seem to render anything unique in the Markdown preview GH generates?

**Comment by nickfrosty** [comment]
@ngundotra can you be more specific about reconciling with the spl page? Are you suggesting we remove one of the docs? Or merge them together? Or something else?

I will have to check with some people about linking to the calculator. But if others agree, then I can add a link to it.

the `:::` syntax is a short hand for display the callout cards within Docusaurus. GitHub does not render anything special for them. You can see an example of what they look like on the [edge docs here](https://edge.docs.solana.com/terminology#lamport))

File: docs/src/developing/guides/compressed-nfts.md [hunk 1]
```diff
@@ -0,0 +1,862 @@
...
+
+Now we should have all the information we need to actually mint our compressed
+NFT.
+
+Using the `createMintToCollectionV1Instruction` helper function provided in the
+Bubblegum SDK, we can craft the instruction to actually mint our compressed NFT
+directly into our collection.
+
+If you have minted traditional NFTs on Solana, this will look fairly similar. We
+are creating a new instruction, giving several several of the account addresses
```
Comment (jacobcreech) [inline docs/src/developing/guides/compressed-nfts.md:442]:
```suggestion
are creating a new instruction, giving several of the account addresses
```

File: docs/src/developing/guides/compressed-nfts.md [hunk 2]
```diff
@@ -0,0 +1,862 @@
...
+
+:::warning Asset ID
+
+Within the Read API, digital assets (i.e. NFTs) are indexed by their `id`. This
+asset `id` value differs slightly between traditional NFTs and compressed NFTs:
+
+- for traditional/uncompressed NFTs: this is the token's address for the actual
+  Account on-chain that stores the metadata for the asset.
+- for compressed NFTs: this is the `id` of the compressed NFT within the tree
+  and is **NOT** an actual on-chain Account address. While a compressed NFTs
```
Comment (jacobcreech) [inline docs/src/developing/guides/compressed-nfts.md:543]:
```suggestion
  and is **NOT** an actual on-chain Account address. While a compressed NFT's
```

File: docs/src/developing/guides/compressed-nfts.md [hunk 3]
```diff
@@ -0,0 +1,862 @@
...
+  `assetId` resembles a traditional Solana Account address, it is not.
+
+:::
+
+### Common Read API Methods
+
+While the Read API supports more than these listed below, the most commonly used
+methods are:
+
+- `getAsset` - get an specific NFT asset by its `id`
```
Comment (jacobcreech) [inline docs/src/developing/guides/compressed-nfts.md:553]:
```suggestion
- `getAsset` - get a specific NFT asset by its `id`
```

File: docs/src/developing/guides/compressed-nfts.md [hunk 4]
```diff
@@ -0,0 +1,862 @@
...
+authority to an address other than its owner.
+
+:::
+
+And once confirmed by the cluster, we will have successfully transferred a
+compressed NFT.
+
+## Example code repository
+
+You can find an example code repository for this developer guide on the the
```
Comment (jacobcreech) [inline docs/src/developing/guides/compressed-nfts.md:861]:
```suggestion
You can find an example code repository for this developer guide on the
```

**Review by jacobcreech — COMMENTED** [review]
Looking good! A few nits with extra words here and there

File: docs/src/developing/guides/compressed-nfts.md (hunk 3 above)
Comment (nickfrosty) [reply docs/src/developing/guides/compressed-nfts.md:553]: updated!

File: docs/src/developing/guides/compressed-nfts.md (hunk 2 above)
Comment (nickfrosty) [reply docs/src/developing/guides/compressed-nfts.md:543]: updates

**Review by jacobcreech — APPROVED** [review]

**Pull request merged**